nomad/job_endpoint_validators.go

```diff
@@ -47,9 +47,59 @@ func (c jobNamespaceConstraintCheckHook) Validate(job *structs.Job) (warnings []
 			)
 		}
 	}
+
+	var disallowedNetworkModes []string
+	for _, tg := range job.TaskGroups {
+		for _, network := range tg.Networks {
+			if !taskValidateNetworkMode(network, ns) {
+				network_mode := "host"
+				if len(network.Mode) > 0 {
+					network_mode = network.Mode
+				}
+				disallowedNetworkModes = append(disallowedNetworkModes, network_mode)
+			}
+		}
+	}
+	if len(disallowedNetworkModes) > 0 {
+		if len(disallowedNetworkModes) == 1 {
+			return nil, fmt.Errorf(
+				"used group network mode %q is not allowed in namespace %q", disallowedNetworkModes[0], ns.Name,
+			)
+
+		} else {
+			return nil, fmt.Errorf(
+				"used group network modes %q are not allowed in namespace %q", disallowedNetworkModes, ns.Name,
+			)
+		}
+	}
+
 	return nil, nil
 }
 
+func taskValidateNetworkMode(network *structs.NetworkResource, ns *structs.Namespace) bool {
+	if ns.Capabilities == nil {
+		return true
+	}
+	network_mode := "host"
```

Comment:
Question to reviewers: Is this needed or should the NetworkResource always have a proper mode set? I know that the default is "host" but is that already set here?

Reply:
We'll have passed thru the `(*NetworkResource).Canonicalize` method by this point, but that doesn't set the default mode. IIRC because of a backwards compatibility concern in the past... probably worth revisiting but what you've got here is fine.